### PR TITLE
Add streaming support to HTTPClient

### PR DESCRIPTION
**Pull request by zenhack**
We'll need this to do the OBMd console stuff.

Note that this is built on top of #1028, which should be merged first.

### CODE REVIEW

**Comment by zenhack** [comment]
Rebased.

File: hil/client/client.py [hunk 1]
```diff
@@ -67,10 +67,21 @@ class HTTPResponse(namedtuple('HTTPResponse', ['status_code',
         The http status code
     headers : dict
         http headers
+    body : iterator
+        An iterator that reads the body of the response in chunks, in a
+        streaming fasion.
```
Comment (naved001) [inline hil/client/client.py]: typo here, fas**h**ion

File: hil/test_common.py [hunk 2]
```diff
@@ -113,7 +113,14 @@ def _make_request(self, method, url, data, params):
             )
             return HTTPResponse(status_code=resp.status_code,
                                 headers=resp.headers,
-                                content=resp.get_data())
+                                # As far as I(zenhack) can tell, flask's test
+                                # client doesn't support streaming, so we just
```
Comment (naved001) [inline hil/test_common.py]: yeah, I searched around for it too and didn't see a way to stream with it.

**Comment by zenhack** [comment]
Cool, fixed the typo, merging.

**Pull request merged**